src/Reanimate/Driver.hs

```diff
@@ -154,4 +157,18 @@ reanimate animation = do
 
 
 guessParameter :: Maybe a -> Maybe a -> a -> a
-guessParameter a b def = fromMaybe def (a `mplus` b)
+guessParameter a b def = fromMaybe def (a <|> b)
+
+
+-- If user specifies exactly one dimension explicitly, calculate the other
+userPreferredDimensions :: Maybe Width -> Maybe Height -> Maybe (Width, Height)
+userPreferredDimensions (Just width) (Just height)  = Just (width, height)
+userPreferredDimensions (Just width) Nothing        = Just (width, makeEven $ width * 9 `div` 16) 
+userPreferredDimensions Nothing      (Just height)  = Just (makeEven $ height * 16 `div` 9, height)
+userPreferredDimensions Nothing      Nothing        = Nothing
+
+-- Avoid ffmpeg failures "height not divisible by 2"
+makeEven :: Int -> Int
```

Comment:
This is kind of ad hoc way of fixing 


<details>
  <summary>this kind of issue</summary>
  <pre>
./boundingbox.hs render -h 101
Animation options:
  fps:    60
  width:  178
  height: 101
  fmt:    mp4
  target: /home/jhrcek/Devel/github.com/Lemmih/reanimate/examples/boundingbox.mp4
Starting render of animation: 5.0
300/300

boundingbox.hs: /usr/bin/ffmpeg -r 60 -i '/tmp/reanimate20029-0/render-%05d.svg' -y '-c:v' libx264 -vf 'fps=60' -preset slow -crf 18 -movflags '+faststart' -progress /tmp/reanimate20029-1.txt -pix_fmt yuv420p /home/jhrcek/Devel/github.com/Lemmih/reanimate/examples/boundingbox.mp4:
ffmpeg version 4.0.5 Copyright (c) 2000-2019 the FFmpeg developers
  built with gcc 8 (GCC)
  configuration: --prefix=/usr --bindir=/usr/bin --datadir=/usr/share/ffmpeg --docdir=/usr/share/doc/ffmpeg --incdir=/usr/include/ffmpeg --libdir=/usr/lib64 --mandir=/usr/share/man --arch=x86_64 --optflags='-O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection' --extra-ldflags='-Wl,-z,relro -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld ' --extra-cflags=' ' --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libvo-amrwbenc --enable-version3 --enable-bzlib --disable-crystalhd --enable-fontconfig --enable-frei0r --enable-gcrypt --enable-gnutls --enable-ladspa --enable-libaom --enable-libass --enable-libbluray --enable-libcdio --enable-libdrm --enable-libjack --enable-libfreetype --enable-libfribidi --enable-libgsm --enable-libmp3lame --enable-nvenc --enable-openal --enable-opencl --enable-opengl --enable-libopenjpeg --enable-libopus --enable-libpulse --enable-librsvg --enable-libsoxr --enable-libspeex --enable-libssh --enable-libtheora --enable-libvorbis --enable-libv4l2 --enable-libvidstab --enable-libvmaf --enable-libvpx --enable-libx264 --enable-libx265 --enable-libxvid --enable-libzvbi --enable-avfilter --enable-avresample --enable-postproc --enable-pthreads --disable-static --enable-shared --enable-gpl --disable-debug --disable-stripping --shlibdir=/usr/lib64 --enable-libmfx --enable-runtime-cpudetect
  libavutil      56. 14.100 / 56. 14.100
  libavcodec     58. 18.100 / 58. 18.100
  libavformat    58. 12.100 / 58. 12.100
  libavdevice    58.  3.100 / 58.  3.100
  libavfilter     7. 16.100 /  7. 16.100
  libavresample   4.  0.  0 /  4.  0.  0
  libswscale      5.  1.100 /  5.  1.100
  libswresample   3.  1.100 /  3.  1.100
  libpostproc    55.  1.100 / 55.  1.100
Input #0, image2, from '/tmp/reanimate20029-0/render-%05d.svg':
  Duration: 00:00:12.00, start: 0.000000, bitrate: N/A
    Stream #0:0: Video: svg, bgra, 178x101, 25 fps, 25 tbr, 25 tbn, 25 tbc
Stream mapping:
  Stream #0:0 -> #0:0 (svg (librsvg) -> h264 (libx264))
Press [q] to stop, [?] for help
[libx264 @ 0x55ebc2629140] height not divisible by 2 (178x101)
Error initializing output stream 0:0 -- Error while opening encoder for output stream #0:0 - maybe incorrect parameters such as bit_rate, rate, width or height
Conversion failed!

CallStack (from HasCallStack):
  error, called at src/Reanimate/Misc.hs:33:17 in reanimate-0.1.9.0-E7jbBq2b4Vp3lEYUjFwIEi:Reanimate.Misc
  </pre>
</details>

ffmep fails if either of width / height is not even number.

Probably more robust solution could be to newtype Width / Height and provide smart constructors that would guarantee that they can only contain even numbers. WDYT?

Reply:
The advantage would be that you could provide more user-friendly message before the invalid parameters were passed to ffmpeg renderer, instead of forcing the user to read through that big error message.

Reply:
I'm good either way.

Reply:
Silently changing the height and width to be even numbers would be fine, imo.